src/app/shared/components/smart-search/smart-search.component.ts

```diff
@@ -62,8 +63,14 @@ export class SmartSearchComponent implements AfterViewInit {
 
   autocompleteSelectionHandler(event: MatAutocompleteSelectedEvent) {
     this.selectionHandler(event.option.value);
+    if (this.addOnSelect) {
+      this.add.emit({
```

Comment:
Here you can reuse `addRole` method. It's the same code, of course inside if statement.